### DISCUSSION TITLE
Can programs developed by egui run on windows server?

### DISCUSSION THREAD

**Discussion by AiHaoChen** [emilk/egui#3938] [Q&A]
I released the examples/hello_world_simple program, exe can run on win10 and win11, and the window will be displayed. But I ran it on Windows Server 2019, and nothing happened after double-clicking to execute it. What should I do, what versions of windows does egui support?

**Comment by ekureina** [comment]
I don't think Windows Server 2019 Supports GUIs in general, it is a Server version of the OS.

**Comment by AiHaoChen** [comment]
> I don't think Windows Server 2019 Supports GUIs in general, it is a Server version of the OS.

Is there any way to know that the program is running but there is no response? How to get the running log?
I tried running the command "hello_world_simple.exe 2 > log.txt" in cmd, but there was nothing in log.txt

**Reply by YgorSouza** [reply]
If you copied the examples and you're running in release mode, it might be due to [this line](https://github.com/emilk/egui/blob/00cd671ad8db895438f66b8995ea05199a9333ed/examples/hello_world/src/main.rs#L1). It is meant to prevent the program from opening a terminal window when you double-click on it, but it also prevents it from outputting anything to the console you called it from, as discussed in <https://github.com/rust-lang/rust/issues/67159>.

**Comment by emilk** [comment]
Try `RUST_LOG=debug`
